src/main/java/org/atlasapi/input/ChannelModelTransformer.java

```
@@ -0,0 +1,194 @@
+import org.atlasapi.media.channel.ChannelNumbering;
+import org.atlasapi.media.entity.Alias;
+import org.atlasapi.media.entity.Image;
+import org.atlasapi.media.entity.MediaType;
+import org.atlasapi.media.entity.Publisher;
+import org.atlasapi.media.entity.RelatedLink;
+import org.atlasapi.media.entity.simple.Channel;
+import org.atlasapi.media.entity.simple.PublisherDetails;
+
+import com.metabroadcast.common.base.Maybe;
+import com.metabroadcast.common.ids.NumberToShortStringCodec;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Iterables;
+import org.joda.time.DateTime;
+import org.joda.time.Duration;
+import org.joda.time.LocalDate;
+
+public class ChannelModelTransformer implements ModelTransformer<Channel, org.atlasapi.media.channel.Channel> {
+
+    private final NumberToShortStringCodec v4Codec;
+    private final ImageModelTranslator imageTranslator;
+
+    public ChannelModelTransformer(
+            NumberToShortStringCodec v4Codec,
+            ImageModelTranslator imageTranslator
+    ) {
+        this.v4Codec = v4Codec;
+        this.imageTranslator = imageTranslator;
+    }
+
+    @Override
+    public org.atlasapi.media.channel.Channel transform(Channel simple) {
+        org.atlasapi.media.channel.Channel.Builder complex = org.atlasapi.media.channel.Channel.builder();
+
+        complex.withUri(simple.getUri());
+        complex.withHighDefinition(simple.getHighDefinition());
+        complex.withRegional(simple.getRegional());
+        complex.withAdult(simple.getAdult());
+        complex.withTitle(simple.getTitle());
+
+        Set<Alias> aliases = simple.getV4Aliases().stream().map(simpleAlias -> new Alias(
+                simpleAlias.getNamespace(),
+                simpleAlias.getValue()
+        )).collect(Collectors.toSet());
+        complex.withAliases(aliases);
+
+        if (simple.getMediaType() != null) {
+            Optional<MediaType> mediaType = MediaType.fromKey(simple.getMediaType().toUpperCase());
```

Don't silently swallow the failure if `MediaType.fromKey()` returns absent. That means the simple model contains an invalid media type string. Throw a descriptive exception and fail the call.